### DISCUSSION TITLE
Have VSCode extension SAP HANA Database Explorer in Positron

### DISCUSSION THREAD

**Discussion by pelisonluis** [posit-dev/positron#4987] [Feedback]
Hello! 
First of all, thanks a lot for this amazing product. I am coming from the 'python side' and I am loving your product. I am really enjoying learning and working more with R now.

In my current company, we are working a lot with Shiny and R. For that, we are trying to use more Positron. 
However, our database is SAP HANA, and looks like we don't have any extension in Positron to connect to it and run SQL queries, like the way that it's possible in VSCode and Eclipse, for example.

This feature of being able to connect to SAP HANA Database would be a game change for us, that are currently using VS Code, Eclipse, Jupyter Notebook, RStudio and Positron, so many IDEs...

**In VSCode, there is an extension called ['SAP HANA Database Explorer'](https://marketplace.visualstudio.com/items?itemName=SAPSE.hana-database-explorer&ssr=false). Can we have it in Positron as well? Or something similar?**

I tried to download this extension and upload to Positron, but it doesn't work.

If there is a way of connecting to SAP HANA database, I would appreciate it. If there is something that I can do to help, I would be glad too.

Thanks again.

**Comment by juliasilge** [comment]
Positron uses extensions from OpenVSX, rather than from the proprietary Microsoft marketplace; you [can read more details about the situation here](https://github.com/posit-dev/positron/wiki#vs-code-extensions). It looks like SAP publishes their extension only to the Microsoft marketplace and not to OpenVSX unfortunately, and I can't seem to find it on GitHub. 😕

I have two recommendations for you:

- Request via the SAP support portal that they publish their extension both to the Microsoft extension and to OpenVSX (check out [this request template provided by OpenVSX](https://github.com/open-vsx/publish-extensions/blob/master/docs/external_contribution_request.md))
- As a workaround for now, download the `.vsix` for the extension from its page on the Microsoft marketplace and [install it manually](https://code.visualstudio.com/docs/editor/extension-marketplace#_install-from-a-vsix)

**Reply by pelisonluis** [reply]
Thank you so much, Julia!

I didn't know about these 2 "marketplaces". I tried the second option you suggested, I could install it, but the extension doesn't work properly. 
I will try to request SAP team about the OpenVSX option.

Thanks!
